lightning/src/ln/channelmanager.rs

```diff
@@ -9114,6 +9158,7 @@ where
 									// downstream chan is closed (because we don't have a
 									// channel_id -> peer map entry).
 									counterparty_opt.is_none(),
+									counterparty_opt.cloned().or(monitor.get_counterparty_node_id()),
```

Comment:
Would it be effectively the same to just pass in `counterparty_opt`? It seems like we won't do anything unless the channel is around